packages/@ember/array/index.ts

```diff
@@ -1892,6 +1898,167 @@ const MutableArray = Mixin.create(EmberArray, MutableEnumerable, {
 /**
 @module ember
 */
+
+type AnyArray<T> = EmberArray<T> | Array<T>;
+
+/**
+ * The final definition of NativeArray removes all native methods. This is the list of removed methods
+ * when run in Chrome 106.
+ */
```

Comment:
Note for posterity/future readers looking at this PR historically—we definitely need this, but we need even more to just get the heck rid of it, since this could in principle change dynamically over time.